### PR TITLE
Added other parameters as launchfile args and encoding selection

### PR DESCRIPTION
**Pull request by Virgile-Dauge**
Hello, I'm not sure why, but previous changes already merged into your master Also appears here :/. 

Simple args adding, with few checks on inputs values : 
- ff5de5b  show_image
- f6b2db2 resolution + default is resolution >3
- 3a5529c framerate, Also added a warning in case of downgraded framerate due to too high frequency for selected resolution (not changing the behaviour, but I think it's always good to let user know when everything is not running as asked). 

Best regards,

### CODE REVIEW

File: src/zed_cpu_ros.cpp [hunk 1]
```diff
@@ -379,6 +381,25 @@ class ZedCameraROS {
...
+		 double max_frame_rate;
+		 std::string reso_str = "";
+		 switch (resolution) {
+ 			case 0: max_frame_rate = 15; reso_str = "2K"; break;
+ 			case 1: max_frame_rate = 30; reso_str = "FHD"; break;
+ 			case 2: max_frame_rate = 60; reso_str = "HD"; break;
+ 			case 3: max_frame_rate = 100; reso_str = "VGA"; break;
+ 		}
+		if(frame_rate > max_frame_rate)
+			ROS_WARN("frame_rate(%fHz) to high for resolution(%s), downgraded to %fHz",frame_rate,reso_str.c_str(),max_frame_rate);
```
Comment (willdzeng) [inline src/zed_cpu_ros.cpp]: to -> too

File: src/zed_cpu_ros.cpp [hunk 2]
```diff
@@ -379,6 +381,25 @@ class ZedCameraROS {
...
+	 * @param      frame_rate   			 The camera frame rate
+	 */
+	 void correctFramerate(int resolution, double &frame_rate){
+		 double max_frame_rate;
+		 std::string reso_str = "";
+		 switch (resolution) {
+ 			case 0: max_frame_rate = 15; reso_str = "2K"; break;
+ 			case 1: max_frame_rate = 30; reso_str = "FHD"; break;
+ 			case 2: max_frame_rate = 60; reso_str = "HD"; break;
+ 			case 3: max_frame_rate = 100; reso_str = "VGA"; break;
```
Comment (willdzeng) [inline src/zed_cpu_ros.cpp:399]: default:
ROS_FATAL("Unknow resolution passed");
return;

**Review by willdzeng — CHANGES_REQUESTED** [review]
Just a few comments, thanks.

**Comment by Virgile-Dauge** [comment]
fixed in 85b416f :)

**Comment by Virgile-Dauge** [comment]
5bc4aee added support for encoding selection. It was useful for [Kalibr](https://github.com/ethz-asl/kalibr) which is not working with brg8 encoding.

File: src/zed_cpu_ros.cpp [hunk 3]
```diff
@@ -370,15 +373,36 @@ class ZedCameraROS {
...
 	 * @param[in]  img_frame_id  The image frame identifier
 	 * @param[in]  t             { parameter_description }
+	 * @param[in]  encoding      image_transport encoding
 	 */
-	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t) {
+	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t, const std::string& encoding) {
 		cv_bridge::CvImage cv_image;
 		cv_image.image = img;
-		cv_image.encoding = sensor_msgs::image_encodings::BGR8;
+		cv_image.encoding = encoding;
```
Comment (willdzeng) [inline src/zed_cpu_ros.cpp:381]: cv_image.encoding = encoding_;

File: src/zed_cpu_ros.cpp [hunk 4]
```diff
@@ -370,15 +373,36 @@ class ZedCameraROS {
 	 * @param      img_pub       The image pub
 	 * @param[in]  img_frame_id  The image frame identifier
 	 * @param[in]  t             { parameter_description }
+	 * @param[in]  encoding      image_transport encoding
 	 */
-	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t) {
+	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t, const std::string& encoding) {
```
Comment (willdzeng) [inline src/zed_cpu_ros.cpp:378]: Can you try change the input to const cv::Mat& img?
And test if it works with a subscriber.

Comment (willdzeng) [reply src/zed_cpu_ros.cpp:378]: Also const std::string& img_frame_id

**Review by willdzeng — CHANGES_REQUESTED** [review]

**Review by willdzeng — APPROVED** [review]

**Comment by willdzeng** [comment]
I am going to address the comments myself.
Thanks for your contribution.

File: src/zed_cpu_ros.cpp (hunk 3 above)
Comment (willdzeng) [reply src/zed_cpu_ros.cpp:381]: Does this even work? I tried on my machine changing encoding to rgb8, the color are wrong, looks like we need to do a conversion on the cv mat itself, the image encoding from the output of zed is always bgr8.

**Pull request merged**